File: 101/exercise_rework/easy_4/what_century.rb
# what_century.rb

def get_suffix(centuries)
  last_two_digits = centuries[-2..-1]
  return 'th' if ('11'..'19').cover?(last_two_digits)
  last_digit = centuries[-1]
  case last_digit
  when '1'
    'st'
  when '2'
    'nd'
  when '3'
    'rd'
  else
    'th'
  end
end

def century(year)
  centuries, remainder = year.divmod(100)
  centuries += 1 if remainder > 0
  "#{centuries}#{get_suffix(centuries.to_s)}"
end

# Examples:
p century(2000) == '20th'
puts century(2001) == '21st'
puts century(1965) == '20th'
puts century(256) == '3rd'
puts century(5) == '1st'
puts century(10103) == '102nd'
puts century(1052) == '11th'
puts century(1127) == '12th'
puts century(11201) == '113th'

File: 210/lesson_3/logic_and_flow/small_problems/easy2/what_century.js
// what_century.js

function century(yr) {
  var centuryNum = Math.ceil(yr / 100);
  return appendCenturySuffix(centuryNum);
}

function appendCenturySuffix(centuryNum) {
  var centuryStr = String(centuryNum);
  var lastTwoDigits = centuryNum % 100;
  var lastDigit = centuryNum % 10;
  if (lastTwoDigits >= 11 && lastTwoDigits <= 13) {
    return centuryStr += 'th';
  } else if (lastDigit === 1) {
    return centuryStr += 'st';
  } else if (lastDigit === 2) {
    return centuryStr += 'nd';
  } else if (lastDigit === 3) {
    return centuryStr += 'rd';
  } else {
    return centuryStr += 'th';
  }
}

console.log(century(2000));        // "20th"
console.log(century(2001));        // "21st"
console.log(century(1965));        // "20th"
console.log(century(256));         // "3rd"
console.log(century(5));           // "1st"
console.log(century(10103));       // "102nd"
console.log(century(1052));        // "11th"
console.log(century(1127));        // "12th"
console.log(century(11201));       // "113th"